tests/environment.py

```diff
@@ -188,6 +188,19 @@ def _docker_wait_for_log(container: str, program: str, regex: str, start_line: i
         return (False, start_line, None)
 
 
+def _device_wait_for_log(device: None, log_path: str, regex: str,
+                         start_line: int, timeout: int) -> bool:
+    """Waits for log mathing regex expression to show up."""
+    device.sendline("tail -n10 -f {}".format(log_path))
```

Comment:
This assumes that the log line we're looking for appears in the last 10 lines. I think we should get the entire file instead. And we can actually make use of start_line here.
```suggestion
    device.sendline("tail -n +{:d} -f {}".format(start_line + 1, log_path))
```
(note that line numbering is 1-based, not 0-based, hence the +1)

Reply:
Fixed in #1358. `start_line` now counted in.